File: src/main/java/sg/com/smartinventory/entity/Review.java
package sg.com.smartinventory.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name="review")
public class Review {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name="id")
  private Long id; //PK
  @Column(name="category")
  private String category;
  @Column(name="review_content")
  private String reviewContent;
  @Column(name="rating")
  private int rating;
  @Column(name="customer_id")
  private int customerId; //FK
  @Column(name="product_id")
  private int productId; //FK
}

File: Ports/TheReviewRoom-FastAPI/app/models/review.py
from sqlmodel import Field, Relationship, SQLModel

# Database model, database table inferred from class name.
class Review(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    category: str
    review_content: str
    rating: int
    address: str
    postal_code: int
    phone_number: int
    email: str
    first_name: str
    first_name: str
    customer_id: int | None = Field(default=None, foreign_key="customer.id")
    product_id: int | None = Field(default=None, foreign_key="product.id")
